dialog/fileicon.go

```diff
@@ -34,24 +30,13 @@ func (i *fileIcon) CreateRenderer() fyne.WidgetRenderer {
 }
 
 func mimeTypeGet(path string) (ext, mimeType, mimeSubType string) {
-	ext = filepath.Ext(path[1:])
-	mimeTypeFull := mime.TypeByExtension(ext)
-	if mimeTypeFull == "" {
-		mimeTypeFull = "text/plain"
-		file, err := os.Open(path)
-		if err == nil {
-			defer file.Close()
-			scanner := bufio.NewScanner(file)
-			if scanner.Scan() && !utf8.Valid(scanner.Bytes()) {
-				mimeTypeFull = "application/octet-stream"
-			}
-		}
-	}
+	uri := storage.NewURI(path)
+	ext = uri.Extension()
+	mimeTypeFull := uri.MimeType()
 
 	mimeTypeSplit := strings.Split(mimeTypeFull, "/")
 	mimeType = mimeTypeSplit[0]
-	mimeSubTypeSplit := strings.Split(mimeTypeSplit[1], ";")
-	mimeSubType = mimeSubTypeSplit[0]
+	mimeSubType = mimeTypeSplit[1]
```

Comment:
I don't know if we discussed before, is it OK to panic if the mime does not contain a "/"?

Reply:
It would definitely make sense to add some safety here.